cdci_data_analysis/analysis/queries.py

```diff
@@ -388,13 +386,17 @@ def test_communication(self, instrument, query_type='Real', logger=None, config=
             debug_message += traceback.format_exc()
 
             query_out.set_failed('dataserver communication test',
-                                 extra_message=e_message,
                                  logger=logger,
                                  sentry_client=sentry_client,
                                  excep=e,
                                  e_message=e_message,
                                  debug_message=debug_message)
 
+        except Exception as e:
+            e_message = f'test of communication with backend (instrument: {instrument.name}, product: {self.name}) failed!\n' + repr(e)
+
+            raise BadRequest(e_message)
```

Comment:
this is probably not a bad request, but a bad internal server error.

Reply:
Which exception is the most appropriate?

Reply:
Maybe we do not have one. Actually I added one here https://github.com/oda-hub/dispatcher-app/blob/5c69e0f02a30bc2ab10a9206a6f1e68f1454f5ce/cdci_data_analysis/analysis/exceptions.py#L25-L27)

Reply:
Would be good to actually use this one.